lib/kubernetes-deploy/kubernetes_resource/custom_resource_definition.rb

```diff
@@ -56,6 +56,11 @@ def prunable?
       prunable == "true"
     end
 
+    def predeployed?
+      predeployed = @definition.dig("metadata", "annotations", "kubernetes-deploy.shopify.io/predeployed")
+      predeployed.nil? || predeployed == "true"
```

Comment:
`predeployed.downcase`

Should we raise if the value isn't in `nil, 'false', 'true'` ?

Reply:
We don't do any of that for `prunable`. Do you need those things implemented there too?

Reply:
Good point. Its probably a good idea to do it in both places, but I won't make it a blocker.

Reply:
`prunable` is explicitly documented as "any other value means false"; restricting to only `nil, 'false', 'true'` would be BC-breaking. I'm also not sure why we would want to be case insensitive.